flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/RecoveredInputChannelTest.java

```diff
@@ -252,10 +255,8 @@ private void testConcurrentReadStateAndProcessAndRelease(boolean isRemote) throw
 						buffer.recycleBuffer();
 						numProcessedStates++;
 					}
-				} catch (IllegalStateException e) {
-					if (!e.getMessage().contains("Queried for a buffer after channel has been closed")) {
-						throw e;
-					}
+				} catch (Throwable t) {
+					assertTrue("The expected exception should only happen in the case of released channel.", verifyRelease && inputChannel.isReleased());
```

Comment:
It's better to not hide an original exception in case of failure (to speed up understanding why the test is failing). Either throw the original exception, or embed it into the failure message (as cause) etc.

Reply:
Makes sense to carry the original exception onto the failure messages.